R/git.R

```diff
@@ -323,85 +323,127 @@ git_clean <- function() {
-  }
-  kv_line("Default Git protocol", git_protocol())
   init_default_branch <- git_cfg_get("init.defaultBranch", where = "global")
-  kv_line("Default initial branch name", init_default_branch)
+  if ("git" %in% tool && "user" %in% scope) {
+    cli::cli_h3("Git global (user)")
+    kv_line("Name", git_cfg_get("user.name", "global"))
+    kv_line("Email", git_cfg_get("user.email", "global"))
+    kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
+    vaccinated <- git_vaccinated()
+    kv_line("Vaccinated", vaccinated)
+    if (!vaccinated) {
+      ui_info("See {ui_code('?git_vaccinate')} to learn more")
+    }
+    kv_line("Default Git protocol", git_protocol())
+    kv_line("Default initial branch name", init_default_branch)
+  }
 
   # github (global / user) -----------------------------------------------------
-  hd_line("GitHub")
   default_gh_host <- get_hosturl(default_api_url())
-  kv_line("Default GitHub host", default_gh_host)
-  pat_sitrep(default_gh_host)
+  if ("github" %in% tool && "user" %in% scope) {
+    cli::cli_h3("GitHub user")
+    kv_line("Default GitHub host", default_gh_host)
+    pat_sitrep(default_gh_host)
+  }
 
   # git and github for active project ------------------------------------------
-  hd_line("Git repo for current project")
+  if ("project" %in% scope) {
 
-  if (!proj_active()) {
-    ui_info("No active usethis project")
-    return(invisible())
-  }
-  kv_line("Active usethis project", proj_get())
-  if (!uses_git()) {
-    ui_info("Active project is not a Git repo")
-    return(invisible())
-  }
+    if (!proj_active()) {
+      ui_info("No active usethis project")
+      return(invisible())
+    }
+    cli::cli_h2(glue("Active usethis project: {ui_value(proj_get())}"))
 
-  # local git config -----------------------------------------------------------
-  if (proj_active() && uses_git()) {
-    local_user <- list(
-      user.name = git_cfg_get("user.name", "local"),
-      user.email = git_cfg_get("user.email", "local")
-    )
-    if (!is.null(local_user$user.name) || !is.null(local_user$user.name)) {
-      ui_info("This repo has a locally configured user")
-      kv_line("Name", local_user$user.name)
-      kv_line("Email", local_user$user.email)
+    if (!uses_git()) {
+      ui_info("Active project is not a Git repo")
+      return(invisible())
+    }
+
+    # current branch -------------------------------------------------------------
+    branch <- tryCatch(git_branch(), error = function(e) NULL)
+    tracking_branch <- if (is.null(branch)) NA_character_ else git_branch_tracking()
+    # TODO: can't really express with kv_line() helper
+    branch <- if (is.null(branch)) "<unset>" else branch
+    tracking_branch <- if (is.na(tracking_branch)) "<unset>" else tracking_branch
+
+    # local git config -----------------------------------------------------------
+    if ("git" %in% tool) {
+      cli::cli_h3("Git local (project)")
+      if (proj_active() && uses_git()) {
```

Comment:
This is an observation on pre-existing code, but this check seems unnecessary, yeah? I don't think there's any way to get here if either of these is false. If you agree, this is another simplification we can make.

Reply:
agreed, done in e1c13ef